app/src/main/java/com/github/factotum_sdp/factotum/ui/auth/BaseAuthViewModel.kt

```diff
@@ -0,0 +1,13 @@
+package com.github.factotum_sdp.factotum.ui.auth
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+abstract class BaseAuthViewModel: ViewModel() {
+
+    abstract val _authResult : MutableLiveData<BaseAuthResult<*>>

```

Comment:
This line has to be removed. Only the public state has to be an abstract val.
The MutableLiveData attribute has to remain private and depends on the subclass implementation.